core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -241,10 +243,17 @@ public void messageRead(final InputStream message) {
         messageReceived = true;
 
         listener.onMessage(call.method.parseRequest(message));
+      } catch (Throwable e) {
+        t = e;
       } finally {
         try {
           message.close();
         } catch (IOException e) {
+          if (t != null) {
+            // TODO(carl-mastrangelo): Maybe log e here.
+            Throwables.propagateIfPossible(t);
```

Comment:
propagateIfPossible is [deprecated](https://github.com/google/guava/blob/08d290bbba90e954dd2873e8e4e3d1cc96229b36/guava/src/com/google/common/base/Throwables.java#L158) in Guava 20 and throwIfUnchecked() is the replacement. That said, the removal will happen in July 2018 so it's not a big deal now.